core/src/main/java/org/elasticsearch/action/admin/indices/cache/clear/ShardClearIndicesCacheRequest.java

```diff
@@ -41,8 +42,8 @@
     ShardClearIndicesCacheRequest() {
     }
 
-    ShardClearIndicesCacheRequest(ShardId shardId, ClearIndicesCacheRequest request) {
-        super(shardId, request);
+    ShardClearIndicesCacheRequest(ClearIndicesCacheRequest request, List<ShardRouting> shards, String nodeId) {
+        super(nodeId, request, shards);
         queryCache = request.queryCache();
```

Comment:
I know it's not related and was that way before, but I think these can be gotten from request? do we need to  have these fields stored and streamed separately?